ios/PurchasesHybridCommon/PurchasesHybridCommonSwift/CommonFunctionality.swift

```diff
@@ -0,0 +1,505 @@
+//
+//  CommonFunctionality.swift
+//  PurchasesHybridCommonSwift
+//
+//  Created by Andrés Boedo on 4/19/22.
+//  Copyright © 2022 RevenueCat. All rights reserved.
+//
+
+import Foundation
+import StoreKit
+import Purchases
+
+
+// todo: rename back to RCCommonFunctionality
+@objc(RCCommonFunctionality) public class CommonFunctionality: NSObject {
+
+    // MARK: properties and configuration
+    @objc public static var simulatesAskToBuyInSandbox: Bool = false
+    @objc public static var appUserID: String { Purchases.shared.appUserID }
+    @objc public static var isAnonymous: Bool { Purchases.shared.isAnonymous }
+
+    @objc public static var proxyURLString: String? {
+        get { Purchases.proxyURL?.absoluteString }
+        set {
+            if let value = newValue {
+                guard let proxyURL = URL(string: value) else {
+                    fatalError("could not set the proxyURL, provided value is not a valid URL: \(value)")
```

Comment:
this was an `NSAssert` and it's now a `fatalError`. This is the same approach we took with `purchases-ios`, and also necessitated the `FatalErrorUtil` thing (which I copy-pasted for this PR) because Nimble doesn't natively work with `fatalError` on apple silicon